### ISSUE TITLE
comma-dangle does not catch dangling commas in object destructuring

### ISSUE THREAD

**Issue by feross** [eslint/eslint#2911]
With this rule:

```json
"comma-dangle": [2, "never"]
```

I expect an error with this code (using object destructuring):

```js
var obj = {}
const {
  prop1,
  prop2,
} = obj
```

For comparison, I do get an error (as expected) with this code (using an object literal):

```js
var prop1 = 1
var prop2 = 2
const obj = {
  prop1,
  prop2,
}
```

**Comment by nzakas** [comment]
Makes sense - destructured objects are a different AST node, so we probably didn't update this rule for it.
